airbyte-cdk/python/airbyte_cdk/sources/utils/stream_helpers.py

```diff
@@ -41,7 +41,4 @@ def get_stream_slice(stream: Stream) -> Optional[Mapping[str, Any]]:
                 sync_mode=SyncMode.full_refresh,
             )
         )
-        try:
-            return next(slices)
-        except StopIteration:
-            return {}
```

Comment:
any insight into why previously we explicitly had this try/except? I don't fully understand what this was trying to do and if there's some edge case this was added for that should be directly tested via a new test case.

Reply:
I'm actually not sure. Looks like it was added [here](https://github.com/airbytehq/airbyte/commit/df72bbdf2ced6e476ece9aa619b0486c559b8465)- @girarda do you remember what case you added this for?

Reply:
Maybe it actually makes more senes to 
```
 try:
            return next(slices)
        except StopIteration:
            return None
```
Instead of removing it all together - right now it doesn't change the behavior either way since it's only used in the method above it, but I think this would be the actual behavior we'd want when using it elsewhere.

Reply:
Maybe it's for the case that someone messed something up in what the stream slices returns, we wanted to address it, but we just addressed it incorrectly (empty slice instead of None)?

E.g. the case in my comment below where CommitCommentReactions' `stream_slices` returns an empty generator